components/timeline/src/Timeline.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, ReactNode, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+
+import '../assets/Timeline.scss';
+
+type TimelineItem = {
+  action: string
+  by?: string
+  date: string
+  description: ReactNode
+}
+
+export type TimelineProps = StandardProps & {
+  /** Items is a compulsory property of the Timeline component.
+  Items consists of an array of item objects, each consisting of an
+  action, the name of the person who performed the action (optional),
+  the date (and time) the action was taken and a description of the action.
+  */
+  items: TimelineItem[]
+};
+
+export const Timeline: FC<TimelineProps> = ({
+  children,
+  classBlock,
+  classModifiers,
+  className,
+  items,
+  ...attrs
+}) => {
+  const classes = classBuilder(
+    'hods-timeline',
+    classBlock,
+    classModifiers,
+    className
+  );
+
+  return (
+      <div className={classes()}>
+        {items.map(e => (
```

Comment:
@daniel-ac-martin - thoughts on the `Timeline` component rendering the `TimelineItem`s, as opposed to `TimelineItem`s rendering themselves?

Reply:
Do you mean something like this:
```
<TimeLine>
  <TimeLine.Event datetime="2021-02-18T01:33:23" title="Took rubbish out" by="Dave">
    The rubbish was <strong>very</strong> smelly.
  </TimeLine.Event>
  <TimeLine.Event datetime="2021-01-30T17:23:54" title="Did the dishes" by="Sally">
    We're almost out of washing up liquid.
  </TimeLine.Event>
</TimeLine>
```

I'm not sure. It's looser but there's something ugly about the items taking ReactNode's.

I don't think it's a concern for the first version though.

Reply:
That's a good point. Although, would you ever want to mix and match or would you want to change all events to a special type?

Definitely something to consider in v2. :-)